Library/Homebrew/cmd/fetch.rb

```diff
@@ -64,7 +65,7 @@ def fetch
   def fetch_bottle?(f)
     return true if ARGV.force_bottle? && f.bottle
     return false unless f.bottle && f.pour_bottle?
-    return false if ARGV.build_from_source? || ARGV.build_bottle?
+    return false if FormulaInstaller.build_formula_from_source?(f) || ARGV.build_bottle?
```

Comment:
This feels weird; it could just use a ARGV or another method in here and save that logic for formula_installer itself?